### PR TITLE
Support for quantized kv cache (compressed-tensors)

### PR DESCRIPTION
**Pull request by dbogunowicz**
## Feature description

Adding the logic for loading  quantized models with additional kv cache quantization, generated using `compressed-tensors` framework.

Notable changes:
- `CompressedTensorsConfig` right now expect to read an optional `kv_cache_scheme` argument. As of the next `compressed-tensors` release, the key contains the information about the properties of the quantized kv cache. 
- Added an interface `BaseKVCacheMethod` and its implementation `CompressedTensorsKVCacheMethod`, to prepare `kv_scale` attribute of `Attention` layer on the model initialization

## Current limitations
Currently, the loading of the kv cache scales happens inside the `load_model` method of the model (in this PR only LLaMa model), which is a bit ugly (we need to copy and paste the helper function into every model to make it read quantized kv cache). Something to think about after the initial round of reviews.

Also `compressed-tensors` assume separate value for `k_scale` and `v_scale`, which is not directly compatible with `vLLM`. I added minimal logic to reconciles the requirement for scales to be recomputed for "kv tensor".

The UX for the user would not change, however, they will be able to automatically load the `compressed-tensors` model with this new feature.

### CODE REVIEW

File: tests/quantization/test_compressed_tensors.py [hunk 1]
```diff
@@ -122,3 +126,45 @@ def test_compressed_tensors_w4a16_marlin24(vllm_runner):
         sampling_params = SamplingParams()
         output = llm.generate("Hello world!", sampling_params=sampling_params)
         assert output
+
+
+def test_compressed_tensors_kv_cache(tmp_path, vllm_runner):
+    # test the inference with the compressed-tensors specific
+    # quantized cache
+    model_path = snapshot_download(
```
Comment (robertgshaw2-neuralmagic) [inline tests/quantization/test_compressed_tensors.py:134]: Instead of having this in code, why not just push a model to the hf_hub?

Comment (dbogunowicz) [reply tests/quantization/test_compressed_tensors.py:134]: Will do it on the final round of changes

File: vllm/model_executor/layers/quantization/kv_cache.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+from vllm.model_executor.layers.quantization.base_config import QuantizeMethodBase, QuantizationConfig
+import torch
+from vllm.utils import print_warning_once
+
+
+class BaseKVCacheMethod(QuantizeMethodBase):
+
```
Comment (mgoin) [inline vllm/model_executor/layers/quantization/kv_cache.py:7]: This class could use a doc string describing the assumption of accepting either a single kv_scale or separate k_scale+v_scale and ending up with a single kv_scale

File: vllm/model_executor/layers/quantization/kv_cache.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                    "Currently we only support per-tensor scaling factor (float)"
+                )
+            layer._kv_scale = kv_scale
+            # TODO: We should potentially move this check elsewhere, to discuss
+            if layer._kv_scale == 1.0 and "e5m2" not in layer.kv_cache_dtype:
+                print_warning_once(
+                    "Using KV cache scaling factor 1.0 for fp8_e4m3. This may "
+                    "cause accuracy issues. Please make sure kv-cache scaling "
+                    "factor is available in the fp8 checkpoint.")
+        del layer.kv_scale
```
Comment (mgoin) [inline vllm/model_executor/layers/quantization/kv_cache.py:66]: Do we always want to remove `layer.kv_scale` here? I think this should be moved up a scope

Comment (dbogunowicz) [reply vllm/model_executor/layers/quantization/kv_cache.py:66]: Nice catch!

**Pull request closed without merging**